### PR TITLE
Docs for Vending Machine contracts

### PR DESCRIPTION
**Pull request by cygnusv**

### CODE REVIEW

File: docs/rfc-2-vending-machine.adoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+toc::[]
+
+== Introduction
+
+The Vending Machine contracts implement a special update protocol to enable
+legacy token holders (NU/KEEP) to wrap their tokens and obtain T tokens 
+according to a fixed, pre-defined ratio. The contracts are completely autonomous
+and will work indefinitely, enabling NU and KEEP token holders to join the 
+T network without needing to buy or sell any assets. Logistically, anyone
+holding NU or KEEP can wrap those assets in order to receive T. They can also
```
Comment (mswilkison) [inline docs/rfc-2-vending-machine.adoc:16]:
```suggestion
holding NU or KEEP can wrap those assets in order to upgrade to T. They can also 
```

File: docs/rfc-2-vending-machine.adoc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+== Introduction
+
+The Vending Machine contracts implement a special update protocol to enable
+legacy token holders (NU/KEEP) to wrap their tokens and obtain T tokens 
+according to a fixed, pre-defined ratio. The contracts are completely autonomous
+and will work indefinitely, enabling NU and KEEP token holders to join the 
+T network without needing to buy or sell any assets. Logistically, anyone
+holding NU or KEEP can wrap those assets in order to receive T. They can also 
+unwrap T in order to go back to the underlying asset.
```
Comment (mswilkison) [inline docs/rfc-2-vending-machine.adoc:17]:
```suggestion
unwrap T in order to downgrade back to the underlying asset. 
```

File: docs/rfc-2-vending-machine.adoc [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+Emits an `Unwrapped(address indexed recipient, uint256 tTokenAmount,
+uint256 wrappedTokenAmount)` event.
+
+=== Conversion rate utilities
+
+==== `conversionToT(uint256 amount) public view returns (uint256 tAmount, uint256 wrappedRemainder)`
+
+Returns the T token amount that's obtained from `amount` wrapped tokens 
+(KEEP/NU), and the `wrappedRemainder` that can't be converted.
```
Comment (mswilkison) [inline docs/rfc-2-vending-machine.adoc:57]:
```suggestion
(KEEP/NU), and the `wrappedRemainder` that can't be upgraded.
```

File: docs/rfc-2-vending-machine.adoc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+==== `conversionFromT(uint256 amount) public view returns (uint256 wrappedAmount, uint256 tRemainder)`
+
+Returns the amount of wrapped tokens (KEEP/NU) that's obtained from unwrapping 
+`amount` T tokens, and the `tRemainder` that can't be converted.
```
Comment (mswilkison) [inline docs/rfc-2-vending-machine.adoc:62]:
```suggestion
`amount` T tokens, and the `tRemainder` that can't be downgraded.
```

**Comment by dougvk** [comment]
@cygnusv can we add this https://www.certik.com/projects/threshold-network to the doc

File: docs/rfc-2-vending-machine.adoc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+The vending machine contracts are not ownable.
+
+=== Upgradeability
+
+The vending machine contracts are not upgradeable.
+
+=== Audits
+
+* CertiK (October 2021)
```
Comment (cygnusv) [inline docs/rfc-2-vending-machine.adoc:85]:
```suggestion
* CertiK (October 2021): https://www.certik.com/projects/threshold-network
```

**Pull request merged**